### PR TITLE
ci: Update the ci pipeline to use github docker.

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -26,58 +28,94 @@ jobs:
...
-      - name: (NonProd - Imagery Config) Create pull request tag
-        if: github.ref != 'refs/heads/master' && github.event_name == 'pull_request'
+      - name: (NonProd - Config) Bundle Config File
+        run: |
+          docker run -v $PWD:$PWD ${DOCKER_AWS_ENV} ${BASEMAPS_CONTAINER} bundle --config $PWD/config --output $PWD/config.json
+
+      - name: (NonProd - Config) Compare To Production
         run: |
-          ./bin/bmc.js import -t pr-${{ github.event.number }} --commit
+          docker run -v $PWD:$PWD ${DOCKER_AWS_ENV} ${BASEMAPS_CONTAINER} import --config $PWD/config.json
```
Comment (blacha) [inline .github/workflows/build.yml:37]: I think we should compare to `config-latest` ? this means we dont need to log in to production to do the compare

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -26,58 +28,94 @@ jobs:
...
         run: |
-          ./bin/bmc.js screenshot -t pr-${{ github.event.number }} --host dev.basemaps.linz.govt.nz
+          docker run -v $PWD:$PWD ${DOCKER_AWS_ENV} ${BASEMAPS_CONTAINER} import --config $PWD/config.json --commit
+
+      - name: (NonProd - Config) Prepare path for glyphs and sprites
+        run: |
+          mkdir -p assets/fonts/
+          mkdir -p assets/sprites/
+
+      - name: (NonProd - Config) Build Glyphs
```
Comment (blacha) [inline .github/workflows/build.yml:49]: these are not specific to nonprod, we are just building the glyphs and sprites.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Wentao-Kuang) [reply .github/workflows/build.yml:37]: This is compare the config.json with the dynamo db data.

File: config/style/topographic.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "LINZ Basemaps": { "type": "vector", "url": "/v1/tiles/topographic/EPSG:3857/tile.json", "attribution": "© 2022 Toitū Te Whenua - CC BY 4.0" },
     "LINZ-Texture-Relief": { "type": "raster", "tiles": ["/v1/tiles/texturereliefshade/EPSG:3857/{z}/{x}/{y}.webp"], "tileSize": 256, "minzoom": 0, "maxzoom": 20 }
   },
-  "sprite": "/sprites/topographic",
+  "sprite": "/v1/sprites/topographic",
   "glyphs": "https://orangemug.github.io/font-glyphs/glyphs/{fontstack}/{range}.pbf",
```
Comment (blacha) [inline config/style/topographic.json:11]: this needs to be `/v1/fonts/`

Comment (Wentao-Kuang) [reply config/style/topographic.json:11]: Do we have all the fonts available for the topographic map now? If so, we should switch this.

File: .github/workflows/build.yml (hunk 1 above)
Comment (blacha) [reply .github/workflows/build.yml:37]: I know but to do that we are logging into production. I don't think a pull request should log into production.

production should always be the same as config-latest

Comment (Wentao-Kuang) [reply .github/workflows/build.yml:37]: Sorry for the confusion, this is just compare the config files with the non-prod dynamodb configs.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -12,72 +12,102 @@ jobs:
...
-          ./bin/bmc.js import -t production --commit
-        env:
-          NODE_ENV: "production"
+          aws s3 cp assets.tar.co s3://${BASEMAPS_BUCKET}/assets-latest.tar.co
+
+      - name: (Prod - Config) Compare to DynamoDB
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+        run: |
+          docker run -v $PWD:$PWD ${DOCKER_AWS_ENV} ${BASEMAPS_CONTAINER} import --config $PWD/config.json
+
```
Comment (Wentao-Kuang) [inline .github/workflows/build.yml:105]: Comment out push to prod dynamodb and leave it for the next release.

**Review by blacha — APPROVED** [review]

**Pull request merged**